### PR TITLE
Fix critical invite admin error

### PR DESCRIPTION
**Pull request by tristanhcole**
**Describe the Pull Request**
An admin should not be able to invite a superadmin.

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by tristanhcole** [comment]
[Admin cannot invite a superadmin](https://trello.com/c/InXw0Nyt/587-admin-cannot-invite-a-superadmin)

File: app/test_app/functional/api/test_auth_api.py [hunk 1]
```diff
@@ -207,23 +207,26 @@ def test_logout_api(test_client, authed_sempo_admin_user):
...
-
-        auth_token = authed_sempo_admin_user.encode_auth_token().decode()
-        register_response = test_client.post('/api/v1/auth/permissions/',
-                                             headers=dict(Authorization=auth_token, Accept='application/json'),
-                                             data=json.dumps(dict(email=email, tier=tier)),
-                                             content_type='application/json', follow_redirects=True)
-        assert register_response.status_code == status_code
+
+@pytest.mark.parametrize("admin_tier, email, invite_tier, status_code", [
+    ("superadmin", "test@test.com", "admin", 201),
```
Comment (michielderoos) [inline app/test_app/functional/api/test_auth_api.py:212]: This is giving a 403. After poking around, I think there might be a problem with `requires_auth` behind this

**Review by michielderoos — CHANGES_REQUESTED** [review]

File: app/test_app/functional/api/test_auth_api.py [hunk 2]
```diff
@@ -207,24 +207,6 @@ def test_logout_api(test_client, authed_sempo_admin_user):
...
-        THEN check the response
-        """
-
-        auth_token = authed_sempo_admin_user.encode_auth_token().decode()
-        register_response = test_client.post('/api/v1/auth/permissions/',
-                                             headers=dict(Authorization=auth_token, Accept='application/json'),
-                                             data=json.dumps(dict(email=email, tier=tier)),
-                                             content_type='application/json', follow_redirects=True)
-        assert register_response.status_code == status_code
-
```
Comment (tristanhcole) [inline app/test_app/functional/api/test_auth_api.py:227]: This is a duplicate of `test_create_permissions_api`

File: app/test_app/functional/api/test_auth_api.py (hunk 1 above)
Comment (tristanhcole) [reply app/test_app/functional/api/test_auth_api.py]: Ah yes, sorry was having issues running tests locally. Fixed in latest commit!

**Review by michielderoos — APPROVED** [review]
Nice! LGTM

**Pull request merged**